core/src/main/java/io/questdb/std/str/DirectUtf8Sink.java

```diff
@@ -109,7 +109,7 @@ public DirectUtf8Sink put(@Nullable Utf8Sequence us) {
 
     @Override
     public DirectUtf8Sink put(byte b) {
-        ascii = false;
+        ascii &= b >= 0;
```

Comment:
i don't think this is correct, the idea was to never call put(byte) outside of non-ascii context. I would rather have assert there.

We should never be dealing with unknown bytes. If we know byte is ascii, we should call putAscii(byte) that exists

Reply:
@bluestreak01 wdyt now?

Reply:
yep, this looks consistent with other utf8 sinks